tests/test_spider.py

```diff
@@ -134,7 +134,33 @@ def test_auto_auth_response(mocker):
     assert urls_are_equal(request.url, expected_url)
     assert request.method == "GET"
     assert request.headers == {}
-    assert not request.callback
+
+
+def test_analyze_urls(mocker):
+    spider = EdxSpider(email=None, password=None)
+    fake_json = {
+        "root": "coursename",
+        "blocks": {
+            "course block": {
+                "display_name": "displayname",
+                "block_id": "course",
+                "student_view_url": "http://localhost:8003/xblock/block-coursename+course+type@course+block@course",
+                "lms_web_url": "http://localhost:8003/courses/course-coursename/jump_to/block-coursename+course+type@course+block@course",
```

Comment:
The JSON parser was specifically looking for the lms_web_url, so no other links would be parsed (it appears that the student_view_url has been invalid this whole time). But I edited the parser to find all urls. You can find that change [here](https://github.com/edx/pa11ycrawler/pull/71/commits/eeb3f0ec39b5687e4da574a0b60f1f7526353532#diff-f4723add7686df0216e2b874519042c6R332). I'm open to suggestions here, since it might not be the best way to find valid urls.